seqr/utils/search/utils.py

```diff
@@ -132,11 +132,11 @@ def _get_variants_for_variant_ids(families, variant_ids, user, dataset_type=None
             parsed_variant_ids[variant_id] = None
 
     if dataset_type:
-        def is_valid(v_id):
-            if dataset_type == Sample.DATASET_TYPE_VARIANT_CALLS:
-                return bool(v_id)
-            return not v_id
-        parsed_variant_ids = {k: v for k, v in parsed_variant_ids.items() if is_valid(v)}
+        parsed_variant_ids = {
+            k: v for k, v in parsed_variant_ids.items()
+            if (dataset_type == Sample.DATASET_TYPE_VARIANT_CALLS and v) or
+               (dataset_type == Sample.DATASET_TYPE_SV_CALLS and not v)
```

Comment:
I am concerned about the precedence of `==` and the `and` operators. It is better to add parentheses around the comparisons. And what happens when dataset_type is not `Sample.DATASET_TYPE_VARIANT_CALLS` and `Sample.DATASET_TYPE_SV_CALLS`?

Reply:
The precedence works as expected here, this is a very straightforward boolean in python:
```
s = 'A'
s == 'A' and False
False
s == 'A' and True
True
s = 'B'
s == 'A' and False
False
s == 'A' and True
False
```

That case can't happen, there are only 2 possible dataset types here. I thought it was more readable to explicitly do `dataset_type == Sample.DATASET_TYPE_SV_CALLS` but `dataset_type != Sample.DATASET_TYPE_VARIANT_CALLS` would do the same thing and not leave that edge case open